src/lib/reasoners/shostak.ml

```diff
@@ -646,8 +676,8 @@ struct
           end
 
       | Term t, ty      -> (* case disable_adts() handled here *)
-        if Expr.const_term t ||
-           List.exists (fun (t,_) -> Expr.const_term t) eq then None
+        if Expr.is_model_term t ||
+           List.exists (fun (t,_) -> Expr.is_model_term t) eq then None
```

Comment:
This changes from the old `Expr.const_term` to the new `Expr.is_model_term` — is that intended?

Reply:
It is. The function `Expr.const_term` isn't exposed anymore and it's not appropriate for model generation.